subprojects/gradle/ci-logger/src/main/kotlin/com/avito/utils/logging/Gradle.kt

```diff
@@ -26,10 +28,35 @@ internal object CILoggerRegistry {
 
 private fun provideLogger(project: Project, loggerName: String): CILogger {
     return if (project.buildEnvironment is BuildEnvironment.CI && !project.buildEnvironment.inGradleTestKit) {
+        val endpoints: List<String>? = project.properties["avito.elastic.endpoints"]?.toString()?.split("|")
+        val indexPattern: String? = project.properties["avito.elastic.indexpattern"]?.toString()
+
+        val elasticLogger = if (endpoints != null && indexPattern != null) {
+            project.logger.info(
+                "ElasticLog: initialized with avito.elastic.endpoints=$endpoints " +
+                    "avito.elastic.indexpattern=$indexPattern"
+            )
+
+            ElasticLog(
+                endpoints = endpoints,
+                indexPattern = indexPattern,
+                buildId = project.envArgs.build.id.toString(),
+                verboseHttpLog = null, // don't enable on production, produces huge logs
+                onError = { msg, e -> project.logger.error(msg, e) }
+            )
+        } else {
+            project.logger.info(
+                "ElasticLog: set 'avito.elastic.endpoints'(| separated urls) " +
+                    "and 'avito.elastic.indexpattern' properties to initialize"
+            )
+            null
```

Comment:
I vote for defensive programming in a configuration.

We've faced problems with similar conditional logic. As a client, I'm ready to say that I want to use elastic and want to check that all requirements are satisfied. It's harder to debug broken features because of accidentally missing parameters.

Reply:
ok, i'll change it